app/configurator/components/dataset-control-section.tsx

```diff
@@ -40,50 +75,93 @@ const DatasetRow = ({
   handleDatasetClick: () => void;
   cube: DataCubeMetadata;
 }) => {
+  const ref = useRef<HTMLDivElement>(null);
   const locale = useLocale();
+  const classes = useStyles();
   const [, dispatch] = useConfiguratorState(isConfiguring);
+  const [flash, setFlash] = useState(false);
 
+  const ee = useEventEmitter();
+  useEffect(() => {
+    const handle: EventEmitterHandler<"dataset-added"> = (ev) => {
+      if (ev.datasetIri === cube.iri) {
+        ref.current?.scrollIntoView({ behavior: "smooth" });
+        setTimeout(() => {
+          setFlash(true);
+        }, 300);
+        setTimeout(() => {
+          setFlash(false);
+        }, 5000);
+      }
+    };
+    ee.on("dataset-added", handle);
+    return () => {
+      ee.removeListener("dataset-added", handle);
+    };
+  });
```

Comment:
Nice technique :) 👍 

I wonder if there's a need to explicitly `useEffect` and clean up event listeners like this, can't you use `useEventEmitter("dataset-added", () => { ... })` (it looks like useEffect and cleanup is already included in the hook)? On maybe I missed somthing 😱

Reply:
Ahah I had forgotten I had done that, since I copy pasted this file from another project. Thanks 😅